### PR TITLE
Mark torture failure as red bot.

### PR DESCRIPTION
**Pull request by jfbastien**

### CODE REVIEW

File: buildbot/build.py [hunk 1]
```diff
@@ -63,10 +63,21 @@
...
   sys.stdout.write('\n@@@BUILD_STEP %s@@@\n' % name)
 
 
+def StepLink(label, url):
+  sys.stdout.write('@@@STEP_LINK@%s@%s@@@' % (label, url))
+
+
+def StepFail(name):
+  """Mark one step as failing, but keep going."""
+  sys.stdout.write('\n@@@STEP_FAILURE %s@@@\n' % name)
```
Comment (flagxor) [inline buildbot/build.py]: This needs to be just @@@STEP_FAILURE@@@

File: buildbot/build.py [hunk 2]
```diff
@@ -255,10 +267,11 @@ def main():
   SyncSexpr()
   BuildLLVM()
   TestLLVM()
-  TortureLLVM()
   InstallLLVM()
   ArchiveLLVM(svn_rev)
+  failed_steps = failed_steps + TortureLLVM()
```
Comment (flagxor) [inline buildbot/build.py]: So what you wan to do is have like a summary step that goes red based on the overall status of the build.
BuildStep('summary')
return failed_steps

File: buildbot/build.py (hunk 1 above)
Comment (jfbastien) [inline buildbot/build.py]: Done.

File: buildbot/build.py (hunk 2 above)
Comment (jfbastien) [inline buildbot/build.py]: Done.

**Comment by flagxor** [comment]
lgtm

**Comment by dschuff** [comment]
Relatedly, I've noticed that when there's a failure, the stage that fails is often green even though the build as a whole is red. See e.g. https://build.chromium.org/p/client.wasm.llvm/builders/linux/builds/727 or https://build.chromium.org/p/client.wasm.llvm/builders/linux/builds/785

**Pull request merged**